idaes/apps/caprese/examples/cstr_rodrigo/cstr_rodrigo_model2.py

```diff
@@ -0,0 +1,174 @@
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Test for Cappresse's module for NMPC.
+"""
+
+from pyomo.environ import (Block, ConcreteModel,  Constraint, Expression,
+                            Set, SolverFactory, Var, value, 
+                            TransformationFactory, TerminationCondition, Param,
+                            exp)
+from pyomo.dae.contset import ContinuousSet
+from pyomo.dae.diffvar import DerivativeVar
+
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+
+__author__ = "Kuan-Han Lin"
+
+
+# See if ipopt is available and set up solver
+if SolverFactory('ipopt').available():
+    solver = SolverFactory('ipopt')
+    solver.options = {'tol': 1e-6,
+                      'mu_init': 1e-8,
+                      'bound_push': 1e-8}
+else:
+    solver = None
+    
+
+def make_model(horizon=10, ntfe=5, ntcp=3, steady=False, bounds=False):
+    #sampling time = 2
+    
+    m = ConcreteModel(name = "CSTR_rodrigo")
+    
+    if steady:
+        m.t = Set(initialize = [1.0])
+    else:
+        m.t = ContinuousSet(bounds=(0, horizon))
+        
+    m.T_ind = Set(initialize = ["T", "Tj"])
+    
+    #Parameters
+    m.Cainb = Param(default=1.0)
+    m.Tinb = Param(default=275.0)
+        
+    m.V = Param(initialize=100)
+    m.UA = Param(initialize=20000 * 60)
+    m.rho = Param(initialize=1000)
+    m.Cp = Param(initialize=4.2)
+    m.Vw = Param(initialize=10)
+    m.rhow = Param(initialize=1000)
+    m.Cpw = Param(initialize=4.2)
+    m.k0 = Param(initialize=4.11e13)
+    m.E = Param(initialize=76534.704)
+    m.R = Param(initialize=8.314472)
+    m.Er = Param(initialize=lambda m: (value(m.E) / value(m.R)))
+    m.dH = Param(initialize=596619.)
+
+    m.F = Param(m.t, mutable=True, default=1.2000000000000000E+02)
+    m.Fw = Param(m.t, mutable=True, default=3.0000000000000000E+01)
+    
+    #States
+    m.Ca = Var(m.t,  initialize=1.60659680385930765667001907104350E-02)
+    m.Tall = Var(m.t, m.T_ind)
+    
+    #Algebraic var
+    m.k = Var(m.t,  initialize=4.70706140E+02)
+    
+    #Controls
+    m.Tjinb = Var(m.t, initialize=250)
+    
+    #Derivative variables
+    if steady:
+        m.Cadot = Var(m.t, initialize = 0.0)
+        m.Talldot = Var(m.t, m.T_ind, initialize = 0.0)
+        m.Cadot.fix()
+        m.Talldot.fix()
+    else:
+        m.Cadot = DerivativeVar(m.Ca, initialize=-3.58709135E+01)

```

Comment:
Same comment as above. Where does this number come from and is this much precision necessary?

Reply:
No, they are not necessary. I will update them with only two decimal places.